src/topic.rs

```diff
@@ -0,0 +1,56 @@
+use std::fmt::{Debug, Display, Formatter};
+use std::sync::Arc;
+
+#[derive(Clone, PartialEq, Eq, PartialOrd, Ord, Hash)]
+#[repr(transparent)]
+pub struct Topic(Arc<String>);
+
+impl AsRef<String> for Topic {
+    fn as_ref(&self) -> &String {
+        self.0.as_ref()
+    }
+}
+impl Debug for Topic {
```

Comment:
Debug can be derived

Reply:
If I don't do it by hand it shows `Topic("topic")` in both normal and alternate formats.   Is there a way to derive it so that it drops the outer `Topic(...)` for `{:#?}`?

Reply:
Could this be done with Display?

Reply:
I want it to use `Debug` format though because it is in the `Debug` output for the `State`.  I wouldn't call the `Display` format in a `Debug` formatter for fields normally to keep it consistent.